lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1255,27 +1295,32 @@ def _process_logging_relation_changed(self, relation: Relation):
         if self._charm.unit.is_leader():
             relation.data[self._charm.app].update(self._promtail_binary_url)
             logger.debug("Saved promtail binary url: %s", self._promtail_binary_url)
-            relation.data[self._charm.app]["endpoints"] = json.dumps(self._endpoints())
```

Comment:
This is a breaking change, and a breaking change that will NOT work with Loki charmed in distributed mode, where writing in unit data bags makes no sense.

Reply:
Use instead https://github.com/canonical/traefik-k8s-operator/blob/a98fcd300b6c009bc5b6814a55096387f59abde2/lib/charms/traefik_k8s/v0/ingress_per_unit.py#L403 from the Loki leader.